packages/widgets/src/lib/components/InferenceWidget/widgets/ConversationalWidget/ConversationalWidget.svelte

```diff
@@ -115,34 +149,41 @@
 		}
 	}
 
-	function isValidOutput(arg: any): arg is Response {
-		return (
-			arg && Array.isArray(arg?.conversation?.generated_responses) && Array.isArray(arg?.conversation?.past_user_inputs)
-		);
-	}
-
-	function parseOutput(body: unknown): {
-		conversation: Conversation;
+	function parseOutput(
+		body: unknown,
+		chat: Message[]
+	): {
+		chat: Message[];
 		output: Output;
 	} {
-		if (isValidOutput(body)) {
-			const conversation = body.conversation;
-			const pastUserInputs = conversation.past_user_inputs;
-			const generatedResponses = conversation.generated_responses;
-			const output = pastUserInputs
-				.filter(
-					(x, i) =>
-						x !== null && x !== undefined && generatedResponses[i] !== null && generatedResponses[i] !== undefined
-				)
-				.map((x, i) => ({
-					input: x ?? "",
-					response: generatedResponses[i] ?? "",
-				}));
-			return { conversation, output };
+		if (Array.isArray(body) && body.length) {
+			const text = body[0]?.generated_text ?? "";
+
+			if (!text.length) {
+				throw new TypeError("Model did not generate a response.");
+			}
+
+			const chatWithOutput = [...chat, { role: "assistant", content: text }];
+
+			const output: Output = Array.from({ length: Math.floor(chatWithOutput.length / 2) }, (_, index) => ({
+				input: chatWithOutput[2 * index].content,
+				response: chatWithOutput[2 * index + 1].content,
+			}));
+
+			return { chat: chatWithOutput, output };
 		}
-		throw new TypeError(
-			"Invalid output: output must be of type <conversation: <generated_responses:Array; past_user_inputs:Array>>"
-		);
+		throw new TypeError("Invalid output: output must be of type Array & non-empty");
+	}
+
+	function extractSpecialTokensMap(tokenizerConfig: TokenizerConfig): SpecialTokensMap {
+		const specialTokensMap = Object.create(null);
+		for (const key of SPECIAL_TOKENS_ATTRIBUTES) {
+			const value = tokenizerConfig[key];
+			if (typeof value === "string") {
+				specialTokensMap[key] = value;
+			}
+		}
+		return specialTokensMap;
 	}
```

Comment:
That logic could be moved to utils (`function pick(obj, keys)`)

cc @coyotte508

Reply:
There is arleady a function `pick` defined in both `package/hub/src/utils.ts` and `package/hub/src/inference.ts`, one could be defined for widgets too

Reply:
This is slightly different from the normal `pick` functions, since we only allow strings to be passed in, but I'm happy to abstract it to a `function pickIf(obj, keys, predicate)` if needed.